### ISSUE TITLE
[Feature request] More connections possible on nodes beside edges and middle

### ISSUE THREAD

**Issue by barbershop9** [juzzlin/Heimer#134]
So i noticed that arrows only have three connections: edges of each side of a node, plus the middle of each side.

This make for a very decent workflow in most cases, but in certain (like the ones I'm gonna describe) it can seem a bit lacking.

Let's take for example this:
![2021-01-14_11-01](https://user-images.githubusercontent.com/76448436/104578504-1b99a980-565b-11eb-8cee-c409d34b7763.png)

Where can be seen bunch of text, separated by newlines, and three nodes "trying" to describe or at least represent some context for each connections...

As already known, there only three connections possible per side of nodes, and in this image, we can see three connection already taken...

But what if someone need to explain "visually" each line of a node using other nodes?

![2021-01-14_11-12](https://user-images.githubusercontent.com/76448436/104578799-716e5180-565b-11eb-85b7-103a8febfc15.png)

This is an attempt at making each lines delimited by newlines in the main node, "explained" per lines.

And here the edited version for more clarity (sorry for the grey-scale colors):

![2021-01-14_11-12_betterarrowconnections](https://user-images.githubusercontent.com/76448436/104578966-a7133a80-565b-11eb-970f-bf1c446afc55.png)

Here we can see a node for each line/part of the main node, where things can be more detailed, and easily understood in a visual manner.

This kind of workflow isn't just good in the example i provided, but can be beneficial for people either needing visual help (line by line explanation, etc) or for researcher/people who learn natural language or machine learning (where mindmap concept like these are used a lots).

That's just my opinion on the matter, so feel free to tell me what you think as the author (of this software) and of course others can also pinch in their views on this. 

This is hopefully not too hard to implement given the default connections system for arrows is already pretty decent, but just need either an optional connections like seen on the last image or other method to get to this same result. I believe this might also strengthen the "simplicity" view/mantra of this software, but that's maybe just me :)

**Comment by juzzlin** [comment]
I can see the problem, yes. Maybe the amount of connection points could be a setting and probably saved with the mind map data.

**Comment by barbershop9** [comment]
yes, that would be amazing if this was implemented :100:

**Comment by l3td33r** [comment]
would look nicer if connections were automatically changed according to their position, like this:
![2021-03-11_15-26-27](https://user-images.githubusercontent.com/75885440/110802467-82fc6000-827e-11eb-991a-605aa4d6206d.png)
(that's some old app, not being updated https://github.com/labyrinth-team/labyrinth)

vs how it works in Heimer:
![2021-03-11_15-26-16](https://user-images.githubusercontent.com/75885440/110802469-8394f680-827e-11eb-9b8a-6c71d44719d0.png)

**Comment by juzzlin** [comment]
Yeah, I'll figure out something. The biggest problem currently is my very limited time.
